File: src/csharp/VRE.Vridge.API.Client/Remotes/Beacons/VridgeServerBeacon.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;
using VRE.Vridge.API.Client.Messages.v3.Discovery;

namespace VRE.Vridge.API.Client.Remotes.Beacons
{
    public class VridgeServerBeacon
    {
        private const long TimeoutMs = 5000;
        private static readonly Stopwatch Timer = Stopwatch.StartNew();

        private readonly Beacon beacon;
        private readonly string endpoint;
        private readonly long timestmapMs;        

        public VridgeServerBeacon(Beacon beacon, string endpoint)
        {
            this.beacon = beacon;
            this.endpoint = endpoint;
            timestmapMs = Timer.ElapsedMilliseconds;
        }

        public Beacon Beacon => beacon;
        public string Endpoint => endpoint;
        public bool IsFresh => timestmapMs + TimeoutMs > Timer.ElapsedMilliseconds;

        public override string ToString()
        {
            return beacon.Role + "|" +
                   beacon.HumanReadableVersion + "|" +
                   beacon.MachineName + "|" +
                   beacon.UserName + "@" +
                   endpoint;
        }
    }
}

File: src/java/VRE.Vridge.API/VRE.Vridge.API.Client/src/main/java/com/riftcat/vridge/api/client/java/remotes/beacons/VridgeServerBeacon.java
package com.riftcat.vridge.api.client.java.remotes.beacons;

import com.riftcat.vridge.api.client.java.proto.Beacon;

import java.net.InetAddress;

public class VridgeServerBeacon {
    private static long timeoutMs = 5000;

    private Beacon beacon;
    private InetAddress endpoint;
    private long timestmapMs;

    public VridgeServerBeacon(Beacon beacon, InetAddress endpoint) {
        this.beacon = beacon;
        this.endpoint = endpoint;
        timestmapMs = System.currentTimeMillis();
    }

    public Beacon getBeacon() {
        return beacon;
    }

    public InetAddress getEndpoint() {
        return endpoint;
    }

    public boolean isFresh(){
        return timestmapMs + timeoutMs > System.currentTimeMillis();
    }

    @Override
    public String toString() {
        return beacon.getRole() + "|" +
                beacon.getHumanReadableVersion() + "|" +
                beacon.getMachineName() + "|" +
                beacon.getUserName() + "@" +
                endpoint.getHostAddress();
    }
}
